File: 1927-sum-game/1927-sum-game.py
class Solution(object):
    def sumGame(self, num):
        n = len(num)
        half = n // 2

        left_sum = right_sum = 0
        left_q = right_q = 0

        # Process both halves
        for i in range(n):
            if num[i] == '?':
                if i < half:
                    left_q += 1
                else:
                    right_q += 1
            else:
                if i < half:
                    left_sum += int(num[i])
                else:
                    right_sum += int(num[i])

        total_questions = left_q + right_q

        # Alice gets the final move when the number of '?' is odd
        if total_questions % 2 != 0:
            return True

        sum_difference = left_sum - right_sum
        question_difference = right_q - left_q

        # False means Bob can make both halves equal
        return 2 * sum_difference != 9 * question_difference

File: 1927-sum-game/1927-sum-game.cpp
class Solution {
public:
    bool sumGame(string num) {
        int n = num.size();
        int half = n / 2;

        int leftSum = 0, rightSum = 0;
        int leftQuestions = 0, rightQuestions = 0;

        // Calculate digit sums and count '?' in both halves
        for (int i = 0; i < n; i++) {
            if (num[i] == '?') {
                if (i < half) {
                    leftQuestions++;
                } else {
                    rightQuestions++;
                }
            } else {
                if (i < half) {
                    leftSum += num[i] - '0';
                } else {
                    rightSum += num[i] - '0';
                }
            }
        }

        int totalQuestions = leftQuestions + rightQuestions;

        // Alice gets the last move, so she can force unequal sums
        if (totalQuestions % 2 != 0) {
            return true;
        }

        int sumDifference = leftSum - rightSum;
        int questionDifference = rightQuestions - leftQuestions;

        // Bob wins only when he can make both halves equal.
        // Multiplication by 2 avoids floating-point division.
        return 2 * sumDifference != 9 * questionDifference;
    }
};